Sources/Shared/Models/ConnectedApp.swift

```diff
@@ -0,0 +1,237 @@
+//
+//  ConnectedApp.swift
+//  VimeoNetworking
+//
+//  Copyright © 2019 Vimeo. All rights reserved.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+/// An enumeration of the supported connected app types.
+/// - facebook: Represents a connection to Facebook.
+/// - linkedin: Represents a connection to LinkedIn.
+/// - twitter: Represents a connection to Twitter.
+/// - youtube: Represents a connection to YouTube.
+@objc public enum ConnectedAppType: Int {
+    case facebook
+    case linkedin
+    case twitter
+    case youtube
+
+    public var stringValue: String {
+        switch self {
+        case .facebook:
+            return String.facebook
+        case .linkedin:
+            return String.linkedin
+        case .twitter:
+            return String.twitter
+        case .youtube:
+            return String.youtube
+        }
+    }
+}
+
+///  An object that encapsulates the scopes necessary for interacting with features like publishing to social platforms
+///  or simulcasting a live stream.
+@objcMembers
+public class ConnectedAppScopes: VIMModelObject {
+
+    /// All scopes required for publishing to a specific social media platform.
+    public var publishToSocial: [String]?
```

Comment:
Is there a finite list of scopes that we can enumerate to make this a bit more type safe? At the moment I'm not even sure how to format this `var` with the different scopes - is it comma separate, spaces etc?

Reply:
Also does it make sense for all of these properties to be publicly writeable as well? I'm not sure how they are going to be used but the less mutability of these models the better.

Reply:
The actual scopes aren't defined in the client app, instead they come from the server. We get them by looking for an interaction on the user object for a platform, `facebook_connected_app`, for example. The returned `VIMInteraction` will contain the scopes needed for the supported "connected app" features.

This class is supposed to encapsulate and store those scopes on a `VIMInteraction` object. 

```ObjC
@interface VIMInteraction : VIMModelObject
…
@property (nonatomic, strong, nullable) ConnectedAppScopes *allScopes;`
```

But I think you're right about the access for these properties though. I'll take a look.